skyportal/handlers/api/source.py

```diff
@@ -1232,20 +1236,20 @@ def post_source(data, user_id, session):
             "one valid group ID that you belong to."
         )
 
-    try:
-        obj = schema.load(data)
-    except ValidationError as e:
-        raise ValidationError(
-            'Invalid/missing parameters: ' f'{e.normalized_messages()}'
-        )
+    if not obj_already_exists:
+        try:
+            obj = schema.load(data)
+        except ValidationError as e:
+            raise ValidationError(
+                'Invalid/missing parameters: ' f'{e.normalized_messages()}'
+            )
+        session.add(obj)
 
     if (ra is not None) and (dec is not None):
         # This adds a healpix index for a new object being created
         obj.healpix = ha.constants.HPX.lonlat_to_healpix(ra * u.deg, dec * u.deg)
 
-    groups = (
-        Group.query_records_accessible_by(user).filter(Group.id.in_(group_ids)).all()
-    )
+    groups = session.scalars(Group.select(user).where(Group.id.in_(group_ids))).all()
```

Comment:
Should we also test that all groups requested by `group_ids` are accessible (i.e., they were indeed loaded by the select)?

Reply:
good idea.